### DISCUSSION TITLE
get signer

### DISCUSSION THREAD

**Discussion by P40l051** [rainbow-me/rainbowkit#355] [Q&A] [answered]
Beginner question:
How to get the signer (and use it to call smart contracts in write mode)?
When i try to get the signer with this function `const signer = provider.getSigner(); `
I receive the following error and i'm not understanding why because usually it works:
_TypeError: provider.getSigner is not a function_

Thank you

**Comment by peterferguson** [comment, marked as answer]
RainbowKit uses `wagmi` under the hood so you can get a signer as you would using that ([checkout the docs](https://wagmi.sh/docs/hooks/useSigner))

```js

import { useContract, useSigner } from 'wagmi'

function App() {
  const { data: signer, isError, isLoading } = useSigner()

  const contract = useContract({
    addressOrName: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
    contractInterface: ensRegistryABI,
    signerOrProvider: signer,
  })
}
```

**Reply by P40l051** [reply]
Thanks!

**Comment by lucasfernandes** [comment]
Hey, I'm sorry to re-open this, but for the latest wagmi version isn't there useSigner hook anymore. How I can achieve this?

**Reply by DanielSinclair** [reply]
This changed with Wagmi v1 and viem. Here is the same API under a new name:
```diff
- const { data: signer } = useSigner()
+ const { data: walletClient } = useWalletClient()
```
